enterprise/frontend/src/metabase-enterprise/license/components/LicenseAndBillingSettings/LicenseAndBillingSettings.tsx

```diff
@@ -104,21 +129,49 @@ const LicenseAndBillingSettings = ({
         )}
       </>
 
-      <SectionHeader>{t`License`}</SectionHeader>
-
-      <SectionDescription>{description}</SectionDescription>
-
-      <LicenseInput
-        invalid={isInvalid}
-        loading={isUpdating}
-        error={error}
-        token={token}
-        onUpdate={updateToken}
-      />
+      {shouldShowLicenseInput && (
+        <>
+          <SectionHeader>{t`License`}</SectionHeader>
+
+          <SectionDescription>{description}</SectionDescription>
+
+          <LicenseInput
+            disabled={is_env_setting}
+            placeholder={is_env_setting ? t`Using ${env_name}` : undefined}
+            invalid={isInvalid}
+            loading={isUpdating}
+            error={error}
+            token={token}
+            onUpdate={updateToken}
+          />
+        </>
+      )}
+
+      {!shouldShowLicenseInput && shouldUpsell && (
```

Comment:
Note that this means we will never show upsell for self-hosted plans as `shouldShowLicenseInput` will always be true.

This may be fine for now as our only two self-hosted offerings (pro & enterprise) we do not want to show up-sell. But in the future if we offer a self-hosted plan somewhere between starter and pro, this condition will no longer work correctly.

Perhaps the condition should be `tokenStatus.isValid && shouldUpsell`?

Reply:
It makes sense, thanks 🙏